core/src/main/scala/org/http4s/Scheme.scala

```diff
@@ -0,0 +1,46 @@
+package org.http4s
+
+import cats.{Order, Show}
+import org.http4s.util.{hashLower, Renderer, Writer}
+
+final class Scheme(val value: String) extends Comparable[Scheme] {
+  override def equals(o: Any) = o match {
+    case that: Scheme => this.value.equalsIgnoreCase(that.value)
+    case _ => false
+  }
+
+  private[this] var hash = 0
+  override def hashCode(): Int = {
+    if (hash == 0) {
+      hash = hashLower(value)
+    }
+    hash
+  }
+
+  override def toString = s"Scheme($value)"
+
+  override def compareTo(other: Scheme): Int =
+    value.compareToIgnoreCase(other.value)
+}
+
+object Scheme {
+  def apply(s: String): Scheme =
```

Comment:
Should be a macro that validates literals.

Reply:
Actually, we need to decide once and for all whether we like macros or StringContexts for this and standardize.